public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -407,43 +418,74 @@ function getAnnotations(state: DashboardSceneState, dsReferencesMapping?: DSRefe
     if (!(layer instanceof dataLayers.AnnotationsDataLayer)) {
       continue;
     }
+    const datasource = getElementDatasource(layer, layer.state.query, 'annotation', undefined, dsReferencesMapping);
+
+    const layerDs = layer.state.query.datasource;
+    if (!layerDs) {
+      throw new Error('Misconfigured AnnotationsDataLayer: Datasource is required for annotations');
+    }
+
```

Comment:
@ivanortegaalba @axelavargas - this code and the code below is under small refactor i propose in https://github.com/grafana/grafana/pull/107390